Update .NET docs (#6707)
getsentry/sentry-docs: src/platforms/dotnet/common/troubleshooting.mdx

## Unhandled exceptions are not captured when using an async `Main` method

Starting with C# 7.1, a program's [`Main` method](https://docs.microsoft.com/dotnet/csharp/fundamentals/program-structure/main-command-line) can be declared either synchronously or asynchronously.
Later, in C# 9, this was further enhanced by use of [top-level statements](https://learn.microsoft.com/dotnet/csharp/fundamentals/program-structure/top-level-statements), which can also be declared either synchronously or asynchronously

> When using either of these features with the Sentry .NET SDK, an unhandled exception may not have been captured if `SentrySdk.Init` was called with a `using` statement.
> Additionally, any associated transaction should have be marked as failed and sent to Sentry - but may not have been when using an async main.

Reviewer: ```suggestion
If `SentrySdk.Init` is called with a `using` statement while using either of these features with the Sentry .NET SDK,   unhandled exceptions might not get captured. Additionally, if you're using an async main, associated transactions might not get marked as failed and sent to Sentry.
```

> These issues were fully resolved as of version 3.31.0 of the Sentry .NET SDK.  If you are using an older version, simply update to the newest release any you can safely use an async main.
> 
> Also, it is no longer required to dispose the result of calling `SentrySdk.Init`.

Reviewer: ```suggestion
These issues have been fully resolved as of version 3.31.0 of the Sentry .NET SDK. To safely use an async main, update your version to the newest release. It's no longer required to dispose of the result of calling `SentrySdk.Init`.
```
